lib/daru/vector.rb

```diff
@@ -170,21 +171,26 @@ def map!(&block)
     #   vecarr = Daru::Vector.new [1,2,3,4], index: [:a, :e, :i, :o]
     #   vechsh = Daru::Vector.new({a: 1, e: 2, i: 3, o: 4})
     def initialize source, opts={}
-      index, source = parse_source(source, opts)
-      set_name opts[:name]
+      if opts[:type] == :category
+        extend Daru::Category
```

Comment:
If we do a `CategoryVector` class it will mean possibly further confusing users since they'll get a new data container just by specifying the index as a `CategoricalIndex`.

Keeping it as `Daru::Vector` and ensuring that the behavior remains similar for both categorical and normal data will make it easier to use and wrap your head around in general.